dart/lib/src/hub.dart

```diff
@@ -0,0 +1,379 @@
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureException' call is a no-op.",
+      );
+    } else if (throwable == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureException called with null parameter.',
+      );
+    } else {
+      final item = _stack.last;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureException(
+            throwable,
+            stackTrace: stackTrace,
+          );
+        } catch (err) {
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing exception : ${throwable}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureException',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  @override
+  Future<SentryId> captureMessage(
+    Message message, {
+    SeverityLevel level,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureMessage' call is a no-op.",
+      );
+    } else if (message == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureMessage called with null parameter.',
+      );
+    } else {
+      final item = _stack.last;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureMessage(
+            message: message,
+            level: level,
+          );
+        } catch (err) {
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing message with id: ${message}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureMessage',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  @override
+  void addBreadcrumb(Breadcrumb breadCrumb) {
+    // TODO: implement addBreadcrumb
+    throw UnimplementedError();
+  }
+
+  @override
+  void bindClient(SentryClient client) {
+    _stack.add(_StackItem(client, _stack.last.scope));
+  }
+
+  @override
+  void clearBreadcrumbs() {
+    // TODO: implement clearBreadcrumbs
+    throw UnimplementedError();
+  }
+
+  @override
+  Hub clone() {
+    // TODO: implement clone
+    throw UnimplementedError();
+  }
+
+  @override
+  void close() {
+    // TODO: implement close
+    throw UnimplementedError();
+  }
+
+  @override
+  void configureScope(ScopeCallback callback) {
+    // TODO: implement configureScope
+    throw UnimplementedError();
+  }
+
+  @override
+  void popScope() {
+    // TODO: implement popScope
+    throw UnimplementedError();
+  }
+
+  @override
+  void pushScope() {
+    // TODO: implement pushScope
+    throw UnimplementedError();
+  }
```

Comment:
I think we can avoid them, as Dart is single-threaded.
cc @bruno-garcia

Reply:
We can have scopes in a single-threaded scenario, it's actually simpler since we don't have the risk of Thread A pushes and Thread B pops and things out of sync. But maybe we leave this out from the first version to simplify things? since we don't really do push and pop scope on Android either, probably better off leaving it out for now and reconsidering this later.

Reply:
yep, let's go simple/easy for now